hack/testing-olm-upgrade/upgrade-common

```diff
@@ -397,15 +397,18 @@ discover_versions(){
   major_version="$(echo $full_version | cut -d'.' -f1)"
   minor_version="$(echo $full_version | cut -d'.' -f2)"
 
-  # we shouldn't run into this yet... maybe for 6.0 but we should like
-  # add the 'Replaces' spec field and use that for the previous_version
-  if [[ $minor_version -eq 0 ]]; then
-    log::info "Will be unable to calculate the previous_version since our minor version is 0"
-    exit 0
+  export version="$(echo $major_version.$minor_version)"
+
+  # Hard rewrite of 5.0.0 to superficial 4.7.0, as replacement for
+  # the breaking point when OpenShift Logging shifted release schema.
+  if [[ $major_version -eq 5 ]] && [[ $minor_version -eq 0 ]]; then
+    # Set this to 4.7 because we need to calculate the latest previous
+    # version prior to 5.0, which is 4.6 for OpenShift Logging.
+    major_version=4
+    minor_version=7
   fi
 
-  export version="$(echo $major_version.$minor_version)"
```

Comment:
I see the export above the if block, but it serves the same purpose as it would if we exported it here after the if block

Reply:
No it is not! The `export version` should be set to `5.0` so that the upgrade bash sourcery can patch the subscription. However `get_latest_previous_version` needs the value `4.7` to calculate 4.6 as the previous version and in turn create the initial subscription.